vpr/src/place/timing_place.h

```diff
@@ -81,6 +81,70 @@ class PlacerCriticalities {
 
     //Set of pins with criticaltites modified by last call to update_criticalities()
     vtr::vec_id_set<ClusterPinId> cluster_pins_with_modified_criticality_;
+
+    //Updates criticalities: incremental V.S. from scratch
+    void incr_update_criticalities(const SetupTimingInfo* timing_info);
+    void recompute_criticalities();
+};
+
+/* Usage
+ * =====
+ * PlacerSetupSlacks returns the clustered netlist connection setup slack used by 
+ * the placer. This also serves to map atom netlist level slack (i.e. on AtomPinIds) 
+ * to the clustered netlist (i.e. ClusterPinIds) used during placement.
+ *
+ * Setup slacks are calculated by calling update_setup_slacks(), which will 
+ * update setup slacks based on the atom netlist connection setup slacks provided by
+ * the passed in SetupTimingInfo. This is done incrementally, based on the modified
+ * connections/AtomPinIds returned by SetupTimingInfo.
+ *
+ * The setup slacks of individual connections can then be queried by calling the 
+ * setup_slack() member function.
+ *
+ * It also supports iterating via pins_with_modified_setup_slack() through the 
+ * clustered netlist pins/connections which have had their setup slacks modified by 
+ * the last call to update_setup_slacks().
+ */
+class PlacerSetupSlacks {
```

Comment:
Explain that these are raw slacks (can be negative, based on the timing constraint specified by the user or the very difficult automatically set timing constraints if no user-specified one exists).